src/main/routing/shd-router.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * shd-router.c
+ *
+ * This component models the upstream (ISP) router from a host's external facing
+ * network interface. The router uses a queue management algorithm to smooth out
+ * packet bursts from fast networks onto slow networks.
+ *
+ *  Created on: Jan 7, 2020
+ *      Author: rjansen
+ */
+
+#include "shadow.h"
+
+
+struct _Router {
```

Comment:
Probably not a reason to deviate from this convention in this PR, but I believe in general names starting with _[A-Z] are reserved. See e.g. https://stackoverflow.com/a/228797, or more authoritatively [7.1.3 Reserved identifiers](https://port70.net/~nsz/c/c11/n1570.html#7.1.3) "All identifiers that begin with an underscore and either an uppercase letter or another underscore are always reserved for any use. "

Reply:
Noted, thanks. Another thing to decide on for the contributors doc ;)